internal/configs/configurator.go

```diff
@@ -859,7 +867,8 @@ func (cnf *Configurator) UpdateEndpointsForVirtualServers(virtualServerExes []*V
 			return fmt.Errorf("Error adding or updating VirtualServer %v/%v: %w", vs.VirtualServer.Namespace, vs.VirtualServer.Name, err)
 		}
 
-		if cnf.isPlus {
+		// if reloads are not enabled, the upstream will not exist in the NGINX Plus, so we need to skip the update.
+		if cnf.isPlus && cnf.isReloadsEnabled {
```

Comment:
It would be worth checking if we can unify the decision making here with `cnf.reload()`

Reply:
let me know what you think of https://github.com/nginxinc/kubernetes-ingress/pull/1788/commits/db061cb5b55bc53f97e072e92e031467c526e8d7

Reply:
all good thanks